src/metabase/pulse/render/body.clj

```diff
@@ -671,66 +634,103 @@
             card-type          (or (series-setting viz-settings group-key :display)
                                    chart-type
                                    (nth default-combo-chart-types idx))
             y-axis-pos         (or (series-setting viz-settings group-key :axis)
                                    (nth (default-y-pos data axis-group-threshold) idx))]
-        {:name          card-name
-         :color         card-color
+        {:name          column-name
+         :cardName      card-name
          :type          card-type
          :data          selected-row-group
-         :yAxisPosition y-axis-pos}))))
+         :yAxisPosition y-axis-pos
+         :column        (second x-cols)
+         :breakoutValue group-key}))))
+
+(defn- axis-row-fns
+  [card data]
+  [(or (ui-logic/mult-x-axis-rowfn card data) #(vector (first %)))
+   (or (ui-logic/mult-y-axis-rowfn card data) #(vector (second %)))])
+
+(defn- card-result->series
+  "Helper function for `render-multiple-lab-chart` that turns a card query result into a series-settings map in the shape expected by `js-svg/combo chart` (and the combo-chart js code)."
+  [idx result]
+  (let [card            (:card result)
+        data            (get-in result [:result :data])
+        display         (:display card)
+        [x-fn y-fn]     (axis-row-fns card data)
+        enforced-type   (if (= display :scalar) :bar display)
+        card-name       (:name card)
+        viz-settings    (:visualization_settings card)
+        joined-rows     (map (juxt x-fn y-fn)
+                             (common/row-preprocess x-fn y-fn (:rows data)))
+        [x-cols y-cols] ((juxt x-fn y-fn) (get-in result [:result :data :cols]))
+        y-axis-position (nth (default-y-pos data axis-group-threshold) idx)]
+    (->>
+     (if (= (count x-cols) 1)
+       (single-x-axis-combo-series enforced-type joined-rows x-cols y-cols viz-settings card-name)
+       (double-x-axis-combo-series enforced-type joined-rows x-cols y-cols viz-settings card-name))
+     (map #(assoc % :yAxisPosition y-axis-position)))))
+
+(defn- render-multiple-lab-chart
+  "When multiple non-scalar cards are combined, render them as a line, area, or bar chart"
+  [render-type card dashcard {:keys [viz-settings]
+                              :as   data}]
+  (let [viz-settings      (merge viz-settings (:visualization_settings dashcard))
+        multi-res         (pu/execute-multi-card card dashcard)
+        ;; multi-res gets the other results from the set of multis.
+        ;; we shove cards and data here all together below for uniformity's sake
+        viz-settings      (set-default-stacked viz-settings card)
+        cards             (cons card (map :card multi-res))
+        multi-data        (cons data (map #(get-in % [:result :data]) multi-res))
+        rowfns            (mapv axis-row-fns cards multi-data)
+        col-seqs          (map :cols multi-data)
+        first-rowfns      (first rowfns)
```

Comment:
You could destructure instead of juggling all these `first`s and `second`s

```
[[x-fn y-fn]] rowfns
[[x-col] [y-col]] ((juxt x-fn y-fn) ...)
```